cmake/FindFCL.cmake

```diff
@@ -17,15 +17,30 @@ find_package(PkgConfig QUIET)
 pkg_check_modules(PC_FCL fcl QUIET)
 
 # Include directories
-find_path(FCL_INCLUDE_DIRS
-    NAMES fcl/collision.h
-    HINTS ${PC_FCL_INCLUDEDIR}
-    PATHS "${CMAKE_INSTALL_PREFIX}/include")
+if (${PC_FCL_VERSION} VERSION_LESS 0.6.0)
+  find_path(FCL_INCLUDE_DIRS
+      NAMES fcl/collision.h
+      HINTS ${PC_FCL_INCLUDEDIR}
+      PATHS "${CMAKE_INSTALL_PREFIX}/include")
+else()
+  find_path(FCL_INCLUDE_DIRS
+      NAMES fcl/narrowphase/collision.h
+      HINTS ${PC_FCL_INCLUDEDIR}
+      PATHS "${CMAKE_INSTALL_PREFIX}/include")
+endif()
```

Comment:
Both of these changes were required for installing FCL 0.6 alongside a ROS installation. Basically, it forces cmake to try to use the pkg-config info before trying out any of the default directories (which, according to the documentation, it should do already, but that wasn't the result I was getting; maybe it works in a newer version of cmake?). With the include paths, if you just specify `fcl/config.h` it was happily finding the ROS includes, compiling as FCL 0.5, then linking to FCL 0.6, triggering some linker sadness. I also had to modify my `CPATH` environment when using alongside ROS, so maybe that is sufficient for the include directory finding. I can test this further once the semester's over.